### PR TITLE
Fix nut extensions

### PR DESCRIPTION
**Pull request by tonicospinelli**
this PR is to solve Issue #3597

### CODE REVIEW

File: src/Composer/PackageManager.php [hunk 1]
```diff
@@ -499,7 +499,7 @@ private function linkConfig($name)
 
         // Check if we have a config file, and if it's readable. (yet)
         $configfilepath = $this->app['resources']->getPath('extensionsconfig/' . $configfilename);
-        if (is_readable($configfilepath)) {
+        if (is_readable($configfilepath) && $this->app['resources']->getRequest('protocol') != 'cli') {
```
Comment (tonicospinelli) [inline src/Composer/PackageManager.php]:
```shell
[Symfony\Component\Routing\Exception\RouteNotFoundException]
  Unable to generate a URL for the named route "fileedit" as such route does not exist.
```
I don't know why, but the `fileedit` route was not defined. Any way, in command line this data not make sense to show :wink:

File: src/Composer/PackageManager.php [hunk 2]
```diff
@@ -499,7 +499,7 @@ private function linkConfig($name)
 
         // Check if we have a config file, and if it's readable. (yet)
         $configfilepath = $this->app['resources']->getPath('extensionsconfig/' . $configfilename);
-        if (is_readable($configfilepath)) {
+        if (is_readable($configfilepath) && !$this->app->isCli()) {
```
Comment (tonicospinelli) [inline src/Composer/PackageManager.php]:
```shell
[Symfony\Component\Routing\Exception\RouteNotFoundException]
  Unable to generate a URL for the named route "fileedit" as such route does not exist.
```
I don't know why, but the `fileedit` route was not defined. Any way, in command line this data not make sense to show :wink:

File: src/Nut/Extensions.php [hunk 3]
```diff
@@ -25,20 +25,20 @@ protected function configure()
      */
     protected function execute(InputInterface $input, OutputInterface $output)
     {
-        if (count($this->app['extend.manager']->messages)) {
-            foreach ($this->app['extend.manager']->messages as $message) {
+        if (count($this->app['extend.manager']->getMessages())) {
+            foreach ($this->app['extend.manager']->getMessages() as $message) {
                 $output->writeln(sprintf('<error>%s</error>', $message));
             }
```
Comment (GawainLynch) [inline src/Nut/Extensions.php]: This part is correct

File: src/Nut/Extensions.php [hunk 4]
```diff
@@ -25,20 +25,20 @@ protected function configure()
...
-            /** @var \Composer\Package\CompletePackageInterface $package */
-            $package = $ext['package'];
-            $rows[] = [$package->getPrettyName(), $package->getPrettyVersion(), $package->getType(), $package->getDescription()];
+        foreach ($allPackages as $type => $packages) {
+            foreach ($packages as $package) {
+                $rows[] = [$package['name'], $package['version'], $package['type'], $package['descrip']];
+            }
         }
 
         $table = $this->getHelper('table');
```
Comment (GawainLynch) [inline src/Nut/Extensions.php]: This part isn't correct, and also breaks the command. Drop that part and we have merge material :grin:

Comment (tonicospinelli) [inline src/Nut/Extensions.php]: this part is show all packages from command line as showing in Backend UI
I don't get your point :confused:

Comment (GawainLynch) [inline src/Nut/Extensions.php]: With just the `->getMessages()` fix, my output is this:
```
+------------------+---------+----------------+-------------------------------------------------------+
| Name             | Version | Type           | Description                                           |
+------------------+---------+----------------+-------------------------------------------------------+
| bolt/boltbb      | v2.0.2  | bolt-extension | A bulletin board extension                            |
| bolt/clientlogin | v2.1.0  | bolt-extension | OAuth login for a Bolt site's clients in the frontend |
| bolt/geoip       | v2.0.2  | bolt-extension | Adds MaxMind GeoIP lookup ability to your templates   |
| bolt/members     | v2.1.1  | bolt-extension | Membership creation and management for Bolt           |
+------------------+---------+----------------+-------------------------------------------------------+
```

With the second part it error with this:
```
  [Symfony\Component\Routing\Exception\RouteNotFoundException]                           
  Unable to generate a URL for the named route "fileedit" as such route does not exist.  
```
And we very deliberately use objects.

Comment (tonicospinelli) [inline src/Nut/Extensions.php]: ow, I the got the same error and send a new commit fixing it :wink:
```shell
+------------------+---------+----------------+----------------------------------------------------+
| Name             | Version | Type           | Description                                        |
+------------------+---------+----------------+----------------------------------------------------+
| bolt/boltforms   | v2.1.1  | bolt-extension | Symfony Forms API for bolt                         |
| bolt/extension   |         | bolt-extension | A description about your extension should go here. |
+------------------+---------+----------------+----------------------------------------------------+
```
Now working good :smile:

**Comment by GawainLynch** [comment]
Rebase merge here: ba701a84f5f3e257d17f3fbc35d147ed8a9c9aea

Thank you, @tonicospinelli :+1:

**Comment by tonicospinelli** [comment]
:+1: nice!!

So, I'll work to get better result to show all packages using objects (although not returns package objects) and open a new PR

**Comment by GawainLynch** [comment]
I'm not understanding what you need to change… What are you seeing that is inconsistent behaviour compared to to the web UI?

**Comment by tonicospinelli** [comment]
because from command line only shows extensions installed from bolt extensions pages, not show local extensions

**Comment by GawainLynch** [comment]
Aaaah! Good call :grin: 

Then the problem shouldn't be solved in Nut, it should be solved in the implementation.

**Comment by tonicospinelli** [comment]
Yeah!! :grin:
So, thinking about it, I'll learn more about this code and make change to get better implementation :wink:

**Comment by GawainLynch** [comment]
Your contributions are always welcome, amigo!

**Pull request closed without merging**